main.cpp

```diff
@@ -19,13 +18,31 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) { }
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	virtual void run(const MatchFinder::MatchResult& Result) {
+	const auto *CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	auto &SM = *Result.SourceManager;
+
+	auto DestTypeString = Lexer::getSourceText(CharSourceRange::getTokenRange(
+                               CastExpr->getLParenLoc().getLocWithOffset(1),
+                               CastExpr->getRParenLoc().getLocWithOffset(-1)),
+                               SM, Result.Context->getLangOpts());
+
+	auto s = ("static_cast<" + DestTypeString + ">(").str();
+	auto Range = CharSourceRange::getCharRange(
+			CastExpr->getLParenLoc(),
+			CastExpr->getSubExprAsWritten()->getBeginLoc());

```

Comment:
Please describe the difference between the use of `getRParenLoc()` and `getSubExprAsWritten()->getBeginLoc()`.

Reply:
getRParenLoc() will point to the closing bracket character in the variable type if consider example

```
int i = (int)f;
```
 it will point to before f.
 

Using getSubExprAsWritten()->getBeginLoc() pointing to the beginning of the variable name (in this case to f).